src/js/d3_svg_cell_renderer.js

```diff
@@ -0,0 +1,207 @@
+var utils = require('./utils');
+var $ = require('jquery');
+var _ = require('underscore');
+
+// TODO: handle accessing config properties cleaner
+
+function D3SVGRuleSet(renderer) {
+	var self = this;
+	self.rule_map = {};
+	self.renderer = renderer;
+
+	self.addRule = function(condition, d3_shape, attrs, z_index) {
+		var rule_id = Object.keys(self.rule_map).length;
+		if (z_index === undefined) {
+			z_index = rule_id;
+		}
+		self.rule_map[rule_id] = new D3SVGRule(rule_id, self.renderer, condition, d3_shape, attrs, z_index);
+		return rule_id;
+	};
+
+	self.removeRule = function(rule_id) {
+		delete self.rule_map[rule_id];
+	}
+
+	self.getRules = function() {
+		// returns a list of lists of rules to render in the given order
```

Comment:
On second thought I think this is actually okay. The alternative is basically to store the rules in sorted order, which is asymptotically less efficient (O(n^2) = insertion sort vs. O(n) = constant insertions + radix sort when you return the rules) and this makes it simpler to remove the rule (just delete it from the map instead of hunting for it in the list). What do you think?